File: problems/3005/solution.dart
class Solution {
  int maxFrequencyElements(List<int> nums) {
    final freq = <int, int>{};
    var best = 0;
    for (final x in nums) {
      final count = (freq[x] ?? 0) + 1;
      freq[x] = count;
      if (count > best) best = count;
    }
    var ans = 0;
    for (final count in freq.values) {
      if (count == best) ans += count;
    }
    return ans;
  }
}

File: problems/3005/solution.php
<?php
class Solution {
    function maxFrequencyElements($nums) {
        $freq = [];
        $best = 0;
        foreach ($nums as $x) {
            $freq[$x] = ($freq[$x] ?? 0) + 1;
            $best = max($best, $freq[$x]);
        }
        $ans = 0;
        foreach ($freq as $count) if ($count === $best) $ans += $count;
        return $ans;
    }
}
?>
